python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,346 @@
+                    yslice = slices[-2]
+                else:
+                    yslice = slices[-1]
+                if isinstance(yslice, slice):
+                    if yslice.start is not None:
+                        y0 += yslice.start
+                else:
+                    raise IndexError("y index must be a slice")
+            xy0 = Point2I(x0, y0)
+            xyF = Point2I(x0+self.array.shape[-1]-1, y0+self.array.shape[-2]-1)
+            self._xy0 = xy0
+            self._bbox = Box2I(xy0, xyF)
+            for singleObj in self.singles:
+                singleObj.setXY0(xy0)
+
+            # Create instances of the single band objects
+            # whose data points to the appropriate slice
+            # of self.array
+            self._updateSingles(singles.imageType)
+        else:
+            # Extract the single band objects and filters
+            if isinstance(singles, OrderedDict):
+                self._filters = list(singles.keys())
+                _singles = list(singles.values())
+            elif singles is not None:
+                self._filters = filters
+                _singles = singles
+            # Set the required attributes
+            self._bbox = _singles[0].getBBox()
+            self._xy0 = _singles[0].getXY0()
+            if _singles is not None and self.filters is not None:
+                self.imageType = type(_singles[0])
+                self._singles = _singles
+            elif singles is None and self.filters is not None:
+                # Attempt to load a set of images
+                self.imageType = imageType
+                self._singles = []
+                for f in self.filters:
+                    if filterKwargs is not None:
+                        for key, value in filterKwargs:
+                            kwargs[key] = value[f]
+                    self._singles.append(imageType(**kwargs))
+            elif singles is None:
+                err = """Currently an OrderedDict
+                         or a list of filters with a
+                         list of lsst.afw.image.Image<X> objects or
+                         kwargs to open a list of new Image<X> objects
+                         is required"""
+                raise NotImplementedError(err)
+
+            assert all([img.getBBox() == self.bbox for img in self.singles])
+            assert all([type(img) == self.imageType for img in self.singles])
+            self._array = np.array([image.array for image in self.singles])
+            self._updateSingles(type(self.singles[0]))
+
+        # Make sure that all of the parameters have been setup appropriately
+        assert isinstance(self._bbox, Box2I)
+        assert isinstance(self.XY0, Point2I)
+
+    @property
+    def array(self):
+        """Data cube array in multiple bands
+        """
+        return self._array
+
+    @property
+    def shape(self):
+        """Shape of the Multiband Object
+        """
+        return self.array.shape
```

Comment:
I'd remove this; requiring the user to call `obj.array.shape` is a bit more verbose but much clearer, and it's another place where I think it'd be confusing to have a method on a multi-band class but not its single-band counterpart.